wire-ios/Wire-iOS/Sources/UserInterface/Conversation/ConversationRootViewController.swift

```diff
@@ -135,20 +138,36 @@ final class ConversationRootViewController: UIViewController {
         return child
     }
 
-    func configure() {
-        guard let conversationViewController = self.conversationViewController else {
-            return
-        }
-        navHeight = navBarContainer.view.heightAnchor.constraint(equalToConstant: defaultNavBarHeight)
-        setupNavigationBarHeight()
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        if traitCollection.userInterfaceStyle != previousTraitCollection?.userInterfaceStyle {
+            // Refresh navigation items to get new buttons with updated colors
+            guard let conversationViewController = self.conversationViewController else { return }
 
-        guard let navigationBarHeight = navHeight else {
-            return
+            navigationItem.rightBarButtonItems = conversationViewController.rightNavigationItems(forConversation: conversation)
+            navigationItem.leftBarButtonItems = conversationViewController.leftNavigationItems(hasUnread: conversation.hasUnreadMessagesInOtherConversations)
         }
+    }
 
-        self.view.backgroundColor = SemanticColors.View.backgroundDefault
+    func configure() {
+
+        guard let conversationViewController = self.conversationViewController else { return }
+
+        // Since we're not using a custom navigation bar container anymore, we need to
+        // properly set up the standard UINavigationBar items. These items were previously
+        // managed by the custom container (navBarContainer).
```

Comment:
Maybe not necessary to refer to something in the past that is no longer relevant?

Reply:
Agreed! Removed.